driver/chardev.c

```diff
@@ -388,10 +437,53 @@ static ssize_t value_store(struct kobject *kobj, struct attribute *attr,
 		container_of(kobj, struct vck5000_device, kobj_aie);
 	uint64_t offset = drv_priv->mem_addr;
 
+  // Variables to write the packet
+  bool is_write = 1;
+  uint64_t timeout_val = 0;
+  uint32_t wr_idx, queue_size, packet_id, header_type, read_completion;
+  void __iomem *pkt;
+
 	if (validate_aie_address(offset, drv_priv) == AIE_ADDR_OK) {
 		kstrtouint(buf, 0, &value);
+
+#ifdef USE_RW32_PKTS
+    // Step 1: Calculate where our packet is going to exist
+    wr_idx = ioread32(drv_priv->admin_queue + QUEUE_WR_PTR_OFFSET);
+    queue_size = ioread32(drv_priv->admin_queue + QUEUE_SIZE_OFFSET);
+    packet_id = wr_idx % queue_size;
+    iowrite32(wr_idx + 1, drv_priv->admin_queue + QUEUE_WR_PTR_OFFSET);
+
+    // Step 2: Initialize the packet 
+    pkt = drv_priv->admin_queue + QUEUE_RING_OFFSET + PKT_SIZE * packet_id;
+    header_type = (AIR_PKT_TYPE_RW32 << 16) | (HSA_PACKET_TYPE_AGENT_DISPATCH << HSA_PACKET_HEADER_TYPE);
+    iowrite32(HSA_PACKET_TYPE_INVALID, pkt + PKT_HEADER_TYPE_OFFSET);
+    iowrite32((uint32_t)(offset & 0xFFFFFFFF),  pkt + PKT_ARG_ADDR_OFFSET       );  // Writing the addr low bits
+    iowrite32((uint32_t)(offset >> 32),         pkt + PKT_ARG_ADDR_OFFSET + 0x4 );  // Writing the addr high bits
+    iowrite32(value,                            pkt + PKT_ARG_ADDR_OFFSET + 0x8 );  // Writing the value, doesn't matter because read
+    iowrite32((uint32_t)is_write,               pkt + PKT_ARG_ADDR_OFFSET + 0xC );  // Writing the is_write
+    iowrite32(header_type, pkt + PKT_HEADER_TYPE_OFFSET); // Writing dispatch header type and rw32 dude
+    iowrite32(0x00000001, pkt + PKT_COMPL_OFFSET); // Writing a 1 to the signal which will mark the completion
+
+    // Step 3: Ring the doorbell and poll on the read pointer
```

Comment:
same here

Reply:
Addressed in [391351c](https://github.com/Xilinx/mlir-air/pull/250/commits/391351c6ed10c4cca742dad0a07c25f1bdb0666c)